### PR TITLE
[marathon] service checks and improvements

### PR DESCRIPTION
**Pull request by gphat**
# What's This Do?

After working on the just-merged etcd plugin for #1235 I wanted to take a look at the Marathon plugin. Looks like the key problem I saw was already fixed in #1240. While I was here I:

* Refactored some of the magic numbers and arrays up into class variables
* Removed unnecessary includes
* Switched to the service check API
* Added the "disk" attribute from the metrics

# Thanks!

I appreciate you taking the time to look at this. If I can do anything to help a merge, let me know!

### CODE REVIEW

**Comment by gphat** [comment]
Note: This is a replacement for #1266. I used git before being caffeinated and seem to've ruined my branch. :)

File: checks.d/marathon.py [hunk 1]
```diff
@@ -20,98 +28,37 @@ def check(self, instance):
...
-            return None
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                message='%s timed out after %s seconds.' % (url, timeout),
+                tags = ["url:{}".format(url)])
+            raise Exception("Timeout when hitting %s" % url)
 
         if r.status_code != 200:
-            self.status_code_event(url, r, aggregation_key)
-            self.warning("Got %s when hitting %s" % (r.status_code, url))
+            self.status_code_event(url, r)
```
Comment (LeoCavaille) [inline checks.d/marathon.py]: This function does not exist anymore?
You may want to use `r.raise_for_status()` in the
```
try: 
     r = requests.get()
     r.raise_for_status()
except timeout:
    <service_check_criticial>
except httperror:
    <service_check_criticial>
else:
    <service_check_ok>`
```
above (don't know for marathon, but other 2XX status may be OK) and you can get rid of this extra if block.

File: checks.d/marathon.py [hunk 2]
```diff
@@ -20,98 +28,37 @@ def check(self, instance):
...
+            raise Exception("Got %s when hitting %s" % (r.status_code, url))
+        else:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK,
+                tags = ["url:{}".format(url)]
```
Comment (LeoCavaille) [inline checks.d/marathon.py]: Also did you see my comment on the previous PR of this URL tag ?

Comment (gphat) [inline checks.d/marathon.py]: Yes, sorry! I meant to ask where you'd end up with multiple URLs and not want the whole thing. I'm assuming anyone using this wold be hitting something on localhost, which means you'd differentiate by either port or path. Where would trimming down to the base url be useful?

File: checks.d/marathon.py (hunk 1 above)
Comment (gphat) [inline checks.d/marathon.py]: I didn't know about that. Thanks!

**Comment by gphat** [comment]
Updated with your feedback. Thanks for dealing with my dumb re-send of this. :P

**Comment by LeoCavaille** [comment]
:+1:

**Pull request merged**